### PR TITLE
Fix#1271 Update VaButtonDropdown

### LINKED ISSUE #1271: Broken button-dropdown demo in docs
chrome-capture

left icon doesn't work, on open icon should change

### PR DESCRIPTION
**Pull request by RVitaly1978**
## Description
close #1271 

changes:
- [x] - add `useSize` hook
- [x] - add `useStateful` hook (update `cStatefulMixin`)
- [x] - update `useEmitProxy` hook (add mapping `event` to `emit`)
- [x] - rewrite `VaButtonDropdown` component with Composition API
- [x] - update locales
- [x] - update examples in docs and demo

chrome-capture

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Improvement/refactoring (non-breaking change that doesn't add any feature but make things better)

### CODE REVIEW

File: packages/ui/src/composables/useEmitProxy.ts [hunk 1]
```diff
@@ -1,21 +1,33 @@
...
-  const createEmits = () => {
-    return events
-  }
+type Events = string | [string, string]
+
+const getEvent = (event: Events) => Array.isArray(event) ? event[0] : event
+const getEmit = (event: Events) => Array.isArray(event) ? event[1] : event
+
+export const useEmitProxy = (events: Events[]) => {
+  const createEmits = () => events.map(event => getEmit(event))
```
Comment (LighthouseKeeperYN) [inline packages/ui/src/composables/useEmitProxy.ts:7]: `  const createEmits = () => events.map(getEmit)`

File: packages/ui/src/composables/useSize.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+      ? getGlobalConfig().components?.[componentName]?.sizesConfig
+      : undefined
+  })
+
+  const sizeComputed = computed<string>(() => {
+    const { defaultSize, sizes } = props.sizesConfig
+    const defaultSizeGlobal = sizesConfigGlobal.value?.defaultSize
+
+    if (!props.size) {
+      return defaultSizeGlobal
```
Comment (LighthouseKeeperYN) [inline packages/ui/src/composables/useSize.ts:73]: ``return `${defaultSizeGlobal || defaultSize}px` ``

**Review by LighthouseKeeperYN — APPROVED** [review]

**Review by asvae — APPROVED** [review]
Awesome. Thanks!

File: packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue [hunk 3]
```diff
@@ -170,25 +212,31 @@ export default class VaButtonDropdown extends mixins(
...
-      .va-dropdown .va-button {
-        font-size: 1.6rem;
+        border-left: none;
       }
     }
 
-    &.va-button-dropdown--large {
-      .va-dropdown .va-button {
-        font-size: 1.7rem;
+    :not(.va-button-group__left-icon) {
```
Comment (asvae) [inline packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue:234]: Can you describe what's the point of this?

File: packages/ui/src/composables/useEmitProxy.ts (hunk 1 above)
Comment (RVitaly1978) [reply packages/ui/src/composables/useEmitProxy.ts:7]: Fixed it

File: packages/ui/src/composables/useSize.ts (hunk 2 above)
Comment (RVitaly1978) [reply packages/ui/src/composables/useSize.ts:73]: Fixed it

File: packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue [hunk 4]
```diff
@@ -31,129 +27,175 @@
...
-    type: String,
-    default: 'medium',
-    validator: (value: string) => {
-      return ['medium', 'small', 'large'].includes(value)
+const { createEmits, createVOnListeners: createListeners } = useEmitProxy(
+  ['click'],
+)
+
+const { createEmits: createMainButtonEmits, createVOnListeners: createMainButtonListeners } = useEmitProxy(
+  [['click', 'main-button-click']],
```
Comment (m0ksem) [inline packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue:95]: I was a bit confused about it. Can we use here object with keys instead?

```suggestion
  [{ listen: 'click', emit: 'main-button-click' }],
```

File: packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue [hunk 5]
```diff
@@ -31,129 +27,175 @@
...
+    const computedClass = computed(() => ({
+      'va-button-dropdown': true,
+      'va-button-dropdown--split': props.split,
+      'va-button-dropdown--normal': props.size === 'normal',
+      'va-button-dropdown--large': props.size === 'large',
+      'va-button-dropdown--small': props.size === 'small',
+    }))
+
+    const computedButtonIcons = computed(() => {
+      const icon = (props.label || slots.label) && !props.leftIcon ? 'icon-right' : 'icon'
```
Comment (m0ksem) [inline packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue:168]:
```suggestion
      const propName = (props.label || slots.label) && !props.leftIcon ? 'icon-right' : 'icon'
```

**Review by m0ksem — APPROVED** [review]
Please, notice my suggestions.

File: packages/ui/src/composables/useStateful.ts [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+ * if `stateful` prop is `false`
+ * Record<any, any> & Record<'modelValue', T>
+ */
+export function useStateful<T, D extends T | undefined> (
+  props: {
+    [key: string]: any,
+    modelValue: T,
+    stateful: boolean,
+  },
+  emit: (event: 'update:modelValue', ...args: any[]) => void,
```
[the remark this answers is not in the document]
Comment (RVitaly1978) [reply packages/ui/src/composables/useStateful.ts:31]: fixed it

File: packages/ui/src/composables/useStateful.ts [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+}
+
+/**
+ * Returns `valueComputed` that is proxy for `modelValue`
+ * if `stateful` prop is `false`
+ * Record<any, any> & Record<'modelValue', T>
+ */
+export function useStateful<T, D extends T | undefined> (
+  props: {
+    [key: string]: any,
```
[the remark this answers is not in the document]
Comment (RVitaly1978) [reply packages/ui/src/composables/useStateful.ts:27]: fixed it

File: packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue (hunk 4 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue:95]: Replaced array with object

File: packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue [hunk 8]
```diff
@@ -31,129 +27,175 @@
...
+      return props.hideIcon ? {} : { [icon]: computedIcon.value }
+    })
+
+    const computedViewStyles = computed(() => ({
```
[the remark this answers is not in the document]
Comment (RVitaly1978) [reply packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue:172]: used lodash `pick`

File: packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue (hunk 5 above)
Comment (RVitaly1978) [reply packages/ui/src/components/va-button-dropdown/VaButtonDropdown.vue:168]: fixed it

**Pull request merged**